File: BasePlusCommissionEmployee.php
<?php
class BasePlusCommissionEmployee extends CommissionEmployee{
    private float $baseSalary;

    public function __construct(string $firstName, string $lastName, string $socialSecurityNumber, float $grossSales, float $commissionRate, float $baseSalary){
        parent::__construct( $firstName, $lastName, $socialSecurityNumber, $grossSales, $commissionRate);
        $this->setBaseSalary($baseSalary);
    }

    public function getBaseSalary(): float{
        return $this->baseSalary;
    }

    public function setBaseSalary(float $newSalary): void{
        if ($newSalary <= 0.0){
            throw new InvalidArgumentException ("Base salary must be greater than 0.oo");
        }else{
            $this->baseSalary = $newSalary;
        }
    }

    public function earnings(): float {
        return $this->baseSalary + parent::earnings();
    }

    public function displayEmployeeDetails(): void {
        parent::displayEmployeeDetails();
        echo "Base Salary: {$this->baseSalary}\n";
       // echo "Total Earnings: {$this->earnings()}\n";
       
    }
}
?>

File: BasePlusCommissionEmployee.java
public class BasePlusCommissionEmployee extends CommissionEmployee{
    private double baseSalary;

    public BasePlusCommissionEmployee(String firstName, String lastName, String socialSecurityNumber, double grossSales, double commissionRate, double baseSalary) {
        super(firstName, lastName, socialSecurityNumber, grossSales, commissionRate);
        setBaseSalary(baseSalary);
}

public double getBaseSalary() {
     return baseSalary; 
}


public void setBaseSalary(double baseSalary) {
    if (baseSalary <= 0.0) throw new IllegalArgumentException("Base salary must be > 0.0");
    this.baseSalary = baseSalary;
}

@Override
public double earnings() { 
    return getBaseSalary() + super.earnings(); 
}

@Override
public String toString() {
    return String.format("%s%nBase Salary: %.2f%nTotal Earnings: %.2f", super.toString(), baseSalary, earnings());
}


}